### PR TITLE
👨‍👦 Add interceptor inheritance to axios

### PR DESCRIPTION
**Pull request by omrilotan**
1. "Inherit" any existing interceptors of global axios instance when creating a new axios instance.
2. Apply new interceptors to existing instances when adding interceptors to global axios instance.

```js
import axios from 'axios'
import axiosInheritance from 'axios-inheritance'

axiosInheritance(axios)
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+defaults: &defaults
+  working_directory: ~/app
+  docker:
+    - image: circleci/node:14
```
Comment (odeadglaz) [inline .circleci/config.yml:4]: ❤️

File: index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const TYPES = ['request', 'response'];
```
Comment (odeadglaz) [inline index.js]: [PETTY] I would const it, and re-use it in your spec

**Review by odeadglaz — APPROVED** [review]
🚀 Don't forget to stabilize the verison

File: index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+module.exports = function axiosInheritance(axios) {
```
Comment (odeadglaz) [inline index.js:6]: Would add JSDocs

File: index.js (hunk 2 above)
Comment (omrilotan) [reply index.js]: I'll do you one better and remove these strings altogether: ab37205

Comment (odeadglaz) [reply index.js]: supreme.

**Pull request merged**